doc/design/game_design_document.md

```diff
@@ -0,0 +1,165 @@
+engage in real-time combat with intelligent bots in interactive environments.
+Players will create characters with unique skills and experience synergy between
+different abilities to defeat their opponents. The game focuses on strategic
+combat, team coordination, and dynamic interactions within the game world.
+
+## Game overview
+
+### Game concept
+
+Project OMG combines strategic depth with fast-paced action in a multiplayer
+setting. Players will create characters with distinct classes and abilities,
+participating in matches where skill synergy plays a crucial role. The
+environment is interactive, with destructible objects that can influence
+gameplay. Matches are short, with no progression save, ensuring each game is a
+fresh experience.
+
+### Genre
+
+2D top-down, multiplayer, match-based combat.
+
+### Platform
+
+The game will be developed for PC initially, with potential expansion to include
+controller support and mobile platforms in the future.
+
+### Target audience
+
+The game is aimed at players aged 12 and above who enjoy strategy and
+multiplayer games, targeting both casual gamers and competitive players looking
+for an engaging multiplayer experience with a mix of PvP (Player vs Player)
+and PvE (Player vs Environment) elements.
+
+## Gameplay mechanics
+
+### Core mechanics
+
+- **Match-based gameplay**: Each game session is independent, with no
+  progression save. Players engage in short, intense matches.
+- **Character creation**: Players can create characters, choosing from various
+  classes and skills.
+- **Skill synergy**: Combining different skills (e.g., fire + poison) can create
+  unique effects and strategies.
+- **PvPvE**: Players compete against each other and AI-controlled bots, adding
+  an extra layer of challenge.
+- **Interactive environment**: The game world includes destructible objects and
+  other interactable elements that players can use to their advantage.
+
+### Game flow
+
+1. **Matchmaking**: Players join a lobby and are matched with others based on
+   skill level.
+2. **Character selection**: Players choose or create their characters, selecting
+   their class and skills.
+3. **Gameplay**: Players engage in combat, using their skills strategically to
+   defeat opponents and interact with the environment.
+4. **End-game**: Statistics are displayed at the end of each match, showing
+   player performance and other metrics.
+
+### Controls
+
+- **Keyboard and mouse**: Primary control method for PC.
+- **Optional controller support**: Future updates may include support for game controllers.
+
+## Game world
+
+### Setting and theme
+
+Project OMG is set in a fantastical world with various themed arenas. Each
+arena offers unique challenges and opportunities for interaction.
+
+### Levels and environments
+
+The game features multiple arenas, each with destructible objects and
+interactive elements that can be used strategically during matches.
+
+## Characters
+
+### Player characters
+
```

Comment:
I think we should give a rough number of expected classes and skills. Maybe even some types/examples of those. But again, let us discuss it through pms first.